classes/Player.py

```diff
@@ -32,14 +35,35 @@ def bullet_shoot(self,bullet_group):
            Bullet(self.rect.midtop,bullet_group)
            self.bullet_sound.play()
 
-    def hit_target(self,target_group):
+    def hit_target(self,target_group,display_surface):
         if pygame.sprite.spritecollide(self,target_group,False):
-            pygame.quit()
-            sys.exit()
+            self.game_over=True
+            game_over_text = f'GameOver!!'
+            
+            while self.game_over:
+                text_surf = self.font.render(game_over_text,True,(255,255,255))
+                text_rect = text_surf.get_rect(center = (window_width / 2, window_height /2))
+                display_surface.blit(text_surf,text_rect)
+                pygame.draw.rect(
+                    display_surface, 
+                    (200,200,200),
+                    text_rect.inflate(30,30), 
+                    width = 1,
+                    border_radius = 3
+                )
+                pygame.display.update()
+                time.sleep(3)
+                break
+        if self.game_over:
+            self.quit_game()
+            
+    def quit_game(self):            
```

Comment:
This is a great way of handling it. I think we can also reuse this method in the main.py where pygame is quit when the game window is closed. 
I appreciate your efforts in making this part of code work!